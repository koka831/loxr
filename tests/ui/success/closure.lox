fun makeCounter() {
  var i = 0;
  fun count() {
    i = i + 1;
    print i;
  }

  return count;
}

print "c1";
var c1 = makeCounter();
c1();
c1();

print "c2";
var c2 = makeCounter();
c2();
c2();
c1();
