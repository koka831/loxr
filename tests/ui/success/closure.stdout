c1
1
2
c2
1
2
3
